modules/scripts/startup-script/files/get_from_bucket.sh

```diff
@@ -50,11 +50,12 @@ stdlib::get_from_bucket() {
 	[[ -d ${dir} ]] || mkdir "${dir}"
 	local attempt=1
 	local max_attempts=10
-	while [[ $attempt -lt $max_attempts ]]; do
+	while [[ $attempt -le $max_attempts ]]; do
 		if stdlib::cmd gsutil -q cp "${url}" "${dir}/${fname}"; then
 			break
 		else
-			stdlib::error "gsutil reported non-zero exit code fetching ${url}.  Retrying (${attempt}/${max_attempts})"
+			stdlib::error "gsutil reported non-zero exit code fetching ${url}.  Retrying (${attempt}/${max_attempts}) with exponential backoff."
+			sleep $((2 ** attempt))
 			((attempt++))
 		fi
 	done
```

Comment:
This case simulates laggy IAM binding replication with a situation where the IAM bindings are known not to exist at all. The test isn't intended to recover just to demonstrate that it will retry when the IAM bindings don't exist.

If you want to see an initial failure that recovers, you can create a VM and manually altering the bindings after it has booted and it enters this retry loop.